src/aequitas/flow/methods/preprocessing/unawareness.py

```diff
@@ -0,0 +1,168 @@
+
+        Returns
+        -------
+        float
+            Correlation ratio value.
+        """
+        cats, freqs = np.unique(categorical_feature, return_counts=True)
+        numeric_mean = np.mean(numeric_feature)
+        sig_y_bar = 0
+        for i in range(len(cats)):
+            category_mean = np.mean(numeric_feature[categorical_feature == cats[i]])
+            sig_y_bar += np.square(category_mean - numeric_mean) * freqs[i]
+        sig_y = np.sum(np.square(numeric_feature - numeric_mean))
+        statistic = np.sqrt(0 if sig_y == 0 else sig_y_bar / sig_y)
+        return statistic
+
+    def _cramerv(self, a: np.ndarray, b: np.ndarray):
+        """This is a measure of the correlation between two categorical columns.
+        Based on the chi-squared metric, the Cramer's V statistic “scales” the
+        chi-squared to be a percentage of its maximum possible variation. It can
+        take values between 0 and 1, with 1 indicating a complete association
+        between the two variables, and a 0 indicating no association. The
+        Cramer's V is a heavily biased estimator and tends to overestimate the
+        strength of the correlation. Therefore, a biased correction is normally
+        applied to the statistic.
+
+        Parameters
+        ----------
+        a : numpy.ndarray
+            First categorical column.
+        b : numpy.ndarray
+            Second categorical column.
+
+        Returns
+        -------
+        float
+            Cramer's V statistic value.
+        """
+        contingency = pd.crosstab(index=[a], columns=[b])
+        chi2 = chi2_contingency(contingency)[0]
+        n = np.sum(contingency.values)
+        r, k = contingency.shape
+        phi2 = chi2 / n
+
+        phi2_corrected = max(0, phi2 - (k - 1) * (r - 1) / (n - 1))
+        r_corrected = r - (r - 1) ** 2 / (n - 1)
+        k_corrected = k - (k - 1) ** 2 / (n - 1)
+
+        statistic = np.sqrt(phi2_corrected / min(r_corrected - 1, k_corrected - 1))
+        return statistic
+
+    def fit(self, X: pd.DataFrame, y: pd.Series, s: Optional[pd.Series]) -> None:
+        """Calculates the correlations between the features and the sensitive 
+        attribute.
+
+        Parameters
+        ----------
+        X : pandas.DataFrame
+            Feature matrix.
+        y : pandas.Series
+            Label vector.
+        s : pandas.Series
+            Protected attribute vector.
+        """
+        self.correlations = pd.Series(index=X.columns)
+
+        for col in X.columns:
+            if X[col].dtype.name == "category":
+                self.correlations[col] = self._cramerv(
+                    s.astype("category").values, X[col].values
+                )
+            else:
+                self.correlations[col] = self._correlation_ratio(
+                    s.astype("category").values, X[col].values
+                )
+
+        self.correlations = self.correlations.sort_values(ascending=False)
```

Comment:
Since there are square roots when calculating the statistics, I assume the correlation value can not be negative, am I correct?

Reply:
yes! the range of values is explained in the correlation functions' descriptions

Reply:
I don't know how to read xD